src/suites/cts/textures/zero_init.spec.ts

```diff
@@ -0,0 +1,1244 @@
+  if (format.indexOf('uint') !== -1) {
+    value = initializedStateAsUint(state);
+  } else if (format.indexOf('sint') !== -1) {
+    value = initializedStateAsSint(state);
+  } else {
+    value = initializedStateAsFloat(state);
+  }
+  return [value, value, value, value];
+}
+
+function initializedStateAsDepth(state: InitializedState): number {
+  switch (state) {
+    case InitializedState.Zero:
+      return 0;
+    case InitializedState.Canary:
+      return 1;
+    case InitializedState.Invalid:
+      return 0.5;
+  }
+}
+
+function initializedStateAsStencil(state: InitializedState): number {
+  switch (state) {
+    case InitializedState.Zero:
+      return 0;
+    case InitializedState.Canary:
+      return 42;
+    case InitializedState.Invalid:
+      return 17;
+  }
+}
+
+function initializedStateAsNormalized(state: InitializedState, bits: number, signed: boolean) {
+  const float = initializedStateAsFloat(state);
+  if (signed) {
+    const min = -Math.pow(2, bits - 1);
+    const max = Math.pow(2, bits - 1) - 1;
+    if (float >= 0) {
+      return Math.round(float * max);
+    } else {
+      return Math.round(Math.abs(float) * min);
+    }
+  } else {
+    const max = Math.pow(2, bits) - 1;
+    return Math.round(float * max);
+  }
+}
+
+function getRequiredTextureUsage(sampleCount: SampleCounts,
+  uninitializeMethod: UninitializeMethod,
+  readMethod: ReadMethod): GPUTextureUsageFlags {
+  let usage: GPUTextureUsageFlags = C.TextureUsage.CopyDst;
+
+  switch (uninitializeMethod) {
+    case UninitializeMethod.Creation:
+      break;
+    case UninitializeMethod.StoreOpClear:
+      usage |= C.TextureUsage.OutputAttachment;
+      break;
+    default:
+      assert(false);
+  }
+
+  switch (readMethod) {
+    case ReadMethod.CopyToBuffer:
+    case ReadMethod.CopyToTexture:
+      usage |= C.TextureUsage.CopySrc;
+      break;
+    case ReadMethod.Sample:
+      usage |= C.TextureUsage.Sampled;
+      break;
+    case ReadMethod.Storage:
+      usage |= C.TextureUsage.Storage;
+      break;
+    case ReadMethod.DepthTest:
+    case ReadMethod.StencilTest:
+      usage |= C.TextureUsage.OutputAttachment;
+      break;
+    default:
+      assert(false);
+  }
+
+  if (sampleCount > 1) {
+    // Copies to multisampled textures are unspecified. We need OutputAttachment to initialize
+    // data in multisampled textures.
+    usage |= C.TextureUsage.OutputAttachment;
+  }
+
+  return usage;
+}
+
+class TextureZeroInitTest extends GPUTest {
+
+  private gpuDataRepresentation!: {[k in InitializedState]: ArrayBuffer};
+
+  constructor(rec: TestCaseRecorder, params: ParamSpec) {
+    super(rec, params);
+  }
+
+  async init(): Promise<void> {
+    await super.init();
+    await this.initGLSL();
+    this.gpuDataRepresentation = await this.getGPUDataRepresentation();
+  }
+
+  get params(): TestParams {
+    return super.params as TestParams;
+  }
+
+  get textureWidth(): number {
+    return 1 << this.params.mipLevelCount;
```

Comment:
possibly we should be able to test NPOT textures

Reply:
Whoops didn't do this yet.